src/main.cpp

```diff
@@ -142,12 +143,7 @@ int main(int argc, char** argv)
     for (int ii=0; ii < args.length(); ii++) {
         QString filename = args[ii];
         if (!filename.isEmpty() && QFile::exists(filename)) {
-            QString password;
-            if (parser.isSet(pwstdinOption)) {
-                static QTextStream in(stdin, QIODevice::ReadOnly);
-                password = in.readLine();
-            }
-            mainWindow.openDatabase(filename, password, parser.value(keyfileOption));
+            mainWindow.openDatabase(filename, parser.value(pwstdinOption), parser.value(keyfileOption));
```

Comment:
I don't think this is working. You just made the --pw-stdin option effectively a --password option. The reason why we read the password from STDIN instead of the command line is that with the latter option your password would appear in the process list (possibly readable by other users) and in your shell history. You should therefore never provide passwords on the command line.

Reply:
I'm trying to get it to work with stdin but it never sends the password at all, lines 143-149 effectively did nothing, I'm not really sure why it doesn't function the way intended, but so far the only method I've found for getting the password to the input box is to do it more directly like I have done.

Reply:
It works on other platforms, but not on Windows.

Reply:
I'm not entirely sure what I should do then. If it's working for other platforms but not for Windows, then I don't know why it's acting the way it is.

Reply:
The problem is that GUI applications on Windows don't have an attached console like *nix applications have.

Reply:
Would something like:
#ifdef Q_OS_WIN
mainWindow.openDatabase(filename, parser.value(pwstdinOption), parser.value(keyfileOption));
#else
mainWindow.openDatabase(filename, parser.value(password), parser.value(keyfileOption));
#endif

be a good enough solution here so all platforms have a working option?